app/models/book.rb

```diff
@@ -1,3 +1,3 @@
 class Book < ActiveRecord::Base
-    belongs_to :category
+  has_and_belongs_to_many :categories, :through => :books_category
```

Comment:
Check out section 2.6 on this page: http://guides.rubyonrails.org/association_basics.html

I don't think you need to specify `:through => :books_category` here or in `category.rb`. I also don't think you need `books_category.rb` at all (though the corresponding database table *is* needed).